sktime/forecasting/trend/_function_forecaster.py

```diff
@@ -0,0 +1,133 @@
+# !/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements FunctionForecaster."""
+
+__author__ = ["benheid"]
+__all__ = ["FunctionForecaster"]
+
+import numpy as np
+import pandas as pd
+from scipy.optimize import curve_fit
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+
+
+class FunctionForecaster(BaseForecaster):
+    """The FunctionForecaster uses a function that is fitted using scipy curve_fit.
+
+    Parameters
+    ----------
+    function: Callable
+        The function that should be fitted and used to make forecasts.
+    initial_params : list
+        The initial parameters of the functions that should be used for fitting
```

Comment:
We should consider adding more information about the kind of Callable we want `function` to be. Perhaps the docstring from [scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.curve_fit.html) would be a good starting point:
> The model function, f(x, …). It must take the independent variable as the first argument and the parameters to fit as separate remaining arguments.

Reply:
I revise the whole Docstring.